pages/checkout/ideal.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <div>
+    <div id="payment-page">
+      <div class="container">
+        <div class="payment-container">
+          <div id="component-container" class="payment"></div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+
+import { ref, onMounted } from 'vue';
+import { AdyenCheckout, Redirect } from '@adyen/adyen-web';
+import { sendPostRequest } from '~/utils/api';
+
+// Reactive references
+const sessionId = ref('');
+const redirectResult = ref('');
+
+// Function to handle the server response
+function handleServerResponse(res, component) {
+  if (res.action) {
```

Comment:
For each component: change the logic, we don't need to check for the res.action, ensure that the redirect works for payment methods that require a redirect (e.g. iDeal)